docs/reference/migration/migrate_query_refactoring.asciidoc

```diff
@@ -59,6 +59,11 @@ Removed `wrapperQueryBuilder(byte[] source, int offset, int length)`. Instead si
 use  `wrapperQueryBuilder(byte[] source)`. Updated the static factory methods in
 QueryBuilders accordingly.
 
+==== QueryStringQueryBuilder
+
+Removed ability to pass in boost value using `field(String field)` method in form e.g. `field^2`.
+Use the `field(String, float)` method instead.
+
```

Comment:
This is a java api change only, you can always specify fields as "field1^5 field2^3" on the rest layer, but when using the java api you now have to do `.field("field1", 5).field("field2", 3)` instead of being able to do `.field("field1^5").field("field2", 3)` which makes no sense from a java point of view. There is a better method that accepts two arguments, which is the way to go. I don't think we need to keep the old method that requires us to parse a string to get out the actual field name and the boost value as part of the java api builder, that's something that should happen only as part of query parsing. I think this clarifies the change and addresses your concerns.